src/Nethermind/Nethermind.Verkle.Tree/VerkleTree.Prover.cs

```diff
@@ -63,6 +63,18 @@ public ExecutionWitness GenerateExecutionWitness(byte[][] keys, out Banderwagon
         return new ExecutionWitness(stemStateDiffList, proof);
     }
 
+    public void UpdateWithPostStateValues(ExecutionWitness executionWitness)
+    {
+        executionWitness.StateDiff.ForEach(stateDiff =>
+        {
+            stateDiff.SuffixDiffs.ForEach(suffixDiffs =>
+            {
+                byte[] key = stateDiff.Stem.Bytes.Append(suffixDiffs.Suffix).ToArray();
```

Comment:
this creates a lot of allocations 
what you can do here is something like this

use stackalloc to allocate a 32 byte array 
in the first loop. then just copy the stem to the first 31 bytes and then in the second loop update the last byte to query the required key